src/moneypenny/moneypenny.py

```diff
@@ -122,16 +130,51 @@ async def execute_order(self, verb: str, dossier: Dict[str,Any]) -> None:
         dossier: Dossier associated with the order for the user.
 
         """
-        verb=verb.strip().lower()
+        verb = verb.strip().lower()
         # We re-read the orders each time; the config map is allowed to change
         await self._read_orders()
         init_containers = self.orders.get(verb)
+        volumes = self.orders.get("volumes")
         if init_containers is None:
             raise NonsensicalOrderError(f"No orders for verb '{verb}'")
-        await self.k8s_client.make_pod(init_containers=init_containers,
-                                       dossier=dossier)
-
-    async def check_completed(self, username: str) -> Bool:
+        if volumes is None:
+            volumes = []
+        username = dossier["token"]["data"]["uid"]
+        logger.info(f"Submitting order '{verb}' for {username}")
+        pull_secret_name = self.config.docker_secret_name
+        self.k8s_client.make_objects(
+            username=username,
+            init_containers=init_containers,
+            volumes=volumes,
+            dossier=dossier,
+            pull_secret_name=pull_secret_name,
+        )
+        tmout = self.config.moneypenny_timeout
+        expiry = datetime.datetime.now() + datetime.timedelta(seconds=tmout)
+        # Wait for order to complete
+        logger.info(f"Awaiting completion for '{verb}': {username}")
+        count = 0
+        while True:
+            count += 1
+            logger.info(f"Checking on {username}: attempt #{count}")
+            finito = await self.check_completed(username)
+            if finito:
+                break
```

Comment:
Instead of using a break, can we just do the cleanup here?  Having a while true with a break is not the best

Reply:
Fixed.  I also made it not a while true.